src/hotspot/share/logging/logAsyncWriter.cpp

```diff
@@ -170,15 +149,30 @@ void AsyncLogWriter::write() {
 
 void AsyncLogWriter::run() {
   while (true) {
+    AsyncLogMap<AnyObj::C_HEAP> snapshot;
```

Comment:
your optimization is great!

I found that you changed RESOURCE_AREA to C_HEAP here. snapshot is still a local variable. Every iteration will destruct it and free up internal nodes in C_HEAP.

Why can't still use RESOURCE_AREA here? 
'ResourceMark rm' before AsyncLogMap<AnyObj::RESOURCE_AREA > snapshot will reset the arena. Comparing to malloc/free on C_HEAP, free up a bunch of nodes is zero-cost using arena.

Reply:
Thanks for noticing this, in an earlier revision this change was necessary. I'm reverting it back to the old resource allocation.